reporter/statsd/statsd_internal_test.go

```diff
@@ -0,0 +1,59 @@
+package statsd
+
+import (
+	"testing"
+	"time"
+
+	"github.com/cactus/go-statsd-client/v5/statsd"
+	"github.com/cactus/go-statsd-client/v5/statsd/statsdtest"
+	"github.com/hamba/statter"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNew(t *testing.T) {
+	s, err := New("127.0.0.1:1234", "test", WithFlushInterval(time.Second), WithFlushBytes(1))
+	require.NoError(t, err)
+	t.Cleanup(func() { _ = s.Close() })
+
+	assert.Implements(t, (*statter.Reporter)(nil), s)
+	assert.Equal(t, time.Second, s.flushInterval)
+	assert.Equal(t, 1, s.flushBytes)
+
+	_, err = New("127.0", "test")
+	assert.Error(t, err)
+}
```

Comment:
Regarding my previous comment, this should probably also test the default values in the case where no `WithXXX` are given. If the cmd package is the one injecting even the defaults, it means it makes no sense to even use `WithXXX` as they might as well directly be arguments, don't you think 🤔 But maybe I'm missing something here.